implementation/src/main/java/io/smallrye/faulttolerance/SynchronousCircuitBreaker.java

```diff
@@ -190,13 +199,27 @@ private boolean isHalfOpenAttemptAllowed() {
         return halfOpenAttempts.get() < config.get(CircuitBreakerConfig.SUCCESS_THRESHOLD, Integer.class);
     }
 
-    private int getRequestCount() {
-        return successCount.get() + failureCount.get();
+    private void reset() {
+        successCount.set(0);
+        halfOpenAttempts.set(0);
+        rollingWindow.clear();
     }
 
-    private void resetCounts() {
-        successCount.set(0);
-        failureCount.set(0);
+    private int getFailureCount() {
```

Comment:
It's probably practical to keep the window small so going through the lit shouldn't be a problem but if we wanted to be faster we could additionally keep counters as we did, just update them also based on the removed value.

Reply:
@michalszynkiewicz Yes, you're right. I think that a window will usually contain dozens of requests which shouldn't be a problem. Also I was thinking that in future we might need to store other attributes of a request, such as timestamp.

Reply:
By the way, how would you distinguish which result to remove if we only have counters? I.e. no ordering info.

Reply:
I meant the list that we have now + counters

Reply:
I see!